_episodes_rmd/08-plot-ggplot2.Rmd

```diff
@@ -56,37 +56,87 @@ ggplot(data = gapminder, mapping = aes(x = gdpPercap, y = lifeExp)) +
-pass `aes` these columns (e.g. `x = gapminder[, "gdpPercap"]`), this is because
-`ggplot` is smart enough to know to look in the **data** for that column!
+It is instructive to break down this line of code piece by piece. *Really* 
+learning the fundamentals now will save hours of debugging later.
+First, there are 2 interesting things to notice:
+
+1. The axes on the plot are **gdpPerCap** and **lifeExp**
+2. They came from `aes(x = gdpPercap, y = lifeExp)`
+
+In previous lessons, we learned to interact with dataframes using this 
+syntax: `gapminder$gdpPercap`. `ggplot` and other tidyverse packages modify 
+the usual rules of R evaluation and save us some typing.
+
+There are 2 more interesting things to notice:
 
-By itself, the call to `ggplot` isn't enough to draw a figure:
+3. `mapping = aes(...)` sets up aesthetics, which will be explained shortly.
+4. `ggplot() + geom_point()` syntax is an example of the grammar of graphics.
 
-```{r}
+The syntax in item 4. is **e1 + e2** where **e1** is a ggplot object or theme, and 
+**e2** is a plot component. Plot components are the graphical layers, so there 
+can be multiple components on any plot.
+
+
+By itself, the call to `ggplot` isn't enough to draw a figure. Only the base
+shows up, which is more like a blank canvas:
+
+```{r lifeExp-vs-gdpPercap-scatter2, message=FALSE}
 ggplot(data = gapminder, mapping = aes(x = gdpPercap, y = lifeExp))
 ```
 
+If we return to the previous example, we can see a layer being added:
+```{r lifeExp-vs-gdpPercap-scatter3, message=FALSE}
+ggplot(data = gapminder, mapping = aes(x = gdpPercap, y = lifeExp)) +
+  geom_point()
+```
+
 We need to tell `ggplot` how we want to visually represent the data, which we
 do by adding a new **geom** layer. In our example, we used `geom_point`, which
 tells `ggplot` we want to visually represent the relationship between **x** and
 **y** as a scatterplot of points:
 
-```{r lifeExp-vs-gdpPercap-scatter2}
-ggplot(data = gapminder, mapping = aes(x = gdpPercap, y = lifeExp)) +
+
+Now we can do something more interesting:
+```{r lifeExp-vs-gdpPercap-scatter4, message=FALSE}
+ggplot(data = gapminder, mapping = aes(x = gdpPercap, y = lifeExp, color = continent)) +
   geom_point()
 ```
 
+Finally, we can talk about `aes`. `aes` is short for "aesthetics," which modify how the 
+plot looks and feels. In this case we told `ggplot` we
+want to plot the "gdpPercap" column of the gapminder data frame on the x-axis, and
+the "lifeExp" column on the y-axis, and color code by continent.
+
+There are many ways to configure aesthetics, and we'll see m ore throughout the hour.
+
+The important thing to know is there are 2 ways to use aesthetics.
+
+1. First way: **Mapping**
+  * This gives us insights about our data. The example above breaks points down by continent
+  * Mapping  configurations are "inherited" by components. We gave `ggplot` aesthetics, and `geom_point` inherited them.
```

Comment:
An additional point in the **Mapping** section could be: 
* We generally pass the names of columns to arguments when we use aesthetics for mapping; this *maps* variables to specific aesthetics of our plot

Reply:
Hi Jeff,

I haven't forgotten about this, I will get to responding to change requests when I have bandwidth. 

Thanks,
Eirian